cookbooks/bcpc-hadoop/recipes/zabbix_web_monitoring.rb

```diff
@@ -0,0 +1,135 @@
+ruby_block "zabbix_web_monitoring" do
+  block do
+    require 'zabbixapi'
+
+    # Make connection to zabbix api url
+    zbx = ZabbixApi.connect(
+      :url => "https://#{node['bcpc']['management']['vip']}" +
+        ":#{node['bcpc']['zabbix']['web_port']}/api_jsonrpc.php",
+      :user => get_config!('zabbix-admin-user'),
+      :password => "#{get_config!('password','zabbix-admin','os')}"
+    )
+    if zbx.nil?
+      Chef::Log.error("Could not connect to Zabbix server")
+      raise "Could not connect to Zabbix server"
+    end
+
+    # Create zabbix host group same as the chef environment name
+    hostgroup_id = zbx.hostgroups.get_id(:name => "#{node.chef_environment}")
+    if hostgroup_id.nil?
+      hostgroup_id = zbx.hostgroups.create(:name => "#{node.chef_environment}")
+
+      # Permission Guests usergroup to read hostgroup_id
+      guests_ugroup_id = zbx.usergroups.get_id(:name => 'Guests')
+      if guests_ugroup_id.nil?
+        Chef::Log.info("Could not find 'Guests' user group in zabbix")
+      else
+        ret = zbx.usergroups.set_perms(
+                :usrgrpid => guests_ugroup_id, :hostgroupids => [hostgroup_id],
+                :permission => 2)
+        if ret.nil?
+          Chef::Log.info("Failed to permission 'Guests' to read #{node.chef_environment}")
+        end
+      end
+    end #if hostgroup_id.nil?
+
+    # Create host for web moniroting
+    host_name="web_monitoring"
+    host_id = zbx.hosts.get_id(:host => "#{host_name}")
+    if host_id.nil?
+      host_id = zbx.hosts.create(
+        :host => "#{host_name}",
+        :interfaces => [{
+          :type => 1, :main => 1, :ip => '127.0.0.1', :dns => '127.0.0.1',
+          :port => 10050, :useip => 0
+        }],
+        :groups => [:groupid => "#{hostgroup_id}"]
+      )
+    end #if host_id.nil?
+
+    # iterate the web scenarios and create_or_update scenarios and related triggers and actions
+    scenarios = node.default[:bcpc][:hadoop][:zabbix][:web_monitoring][:scenarios] or {}
```

Comment:
This should always be defined thanks to the attributes file right? (Is it necessary to have the `or {}` or could that mask a coding error?